### PR TITLE
Ensure Node Shutdown doesn't stall when all nodes in the cluster have a copy of a shard

### PR DESCRIPTION
**Pull request by AthenaEryma**
This PR fixes the issue with Node Shutdown where it will enter the `STALLED` status both of the following are true:
- There is a copy of the shard on every node in the cluster.
- The node that's shutting down has the primary copy of that shard.

This PR adjusts the behavior of the Get Node Shutdown Status API so that in this situation, the node will
be marked as `COMPLETE` rather than `STALLED`.

### CODE REVIEW

File: x-pack/plugin/shutdown/src/internalClusterTest/java/org/elasticsearch/xpack/shutdown/NodeShutdownShardsIT.java [hunk 1]
```diff
@@ -137,6 +145,70 @@ public void testShardStatusIsCompleteOnNonDataNodes() throws Exception {
         assertThat(getResp.getShutdownStatuses().get(0).migrationStatus().getStatus(), equalTo(COMPLETE));
     }
 
+    // GWB> RENAME THIS TEST AND ALSO ADD JAVADOC
```
Comment (dakrone) [inline x-pack/plugin/shutdown/src/internalClusterTest/java/org/elasticsearch/xpack/shutdown/NodeShutdownShardsIT.java:148]: I think this still needs to be resolved?

File: x-pack/plugin/shutdown/src/internalClusterTest/java/org/elasticsearch/xpack/shutdown/NodeShutdownShardsIT.java [hunk 2]
```diff
@@ -137,6 +145,70 @@ public void testShardStatusIsCompleteOnNonDataNodes() throws Exception {
...
+        AcknowledgedResponse putShutdownResponse = client().execute(PutShutdownNodeAction.INSTANCE, putShutdownRequest).get();
+        assertTrue(putShutdownResponse.isAcknowledged());
+
+        assertBusy(() -> {
+            GetShutdownStatusAction.Response getResp = client().execute(
+                GetShutdownStatusAction.INSTANCE,
+                new GetShutdownStatusAction.Request(nodeToStopId)
+            ).get();
+
+            assertThat(getResp.getShutdownStatuses().get(0).migrationStatus().getStatus(), equalTo(COMPLETE));
```
Comment (dakrone) [inline x-pack/plugin/shutdown/src/internalClusterTest/java/org/elasticsearch/xpack/shutdown/NodeShutdownShardsIT.java:180]: Do we need an additional test that we still return `STALLED` when some indices have 0 replicas on the original index, or do we have that elsewhere already?

File: x-pack/plugin/shutdown/src/main/java/org/elasticsearch/xpack/shutdown/TransportGetShutdownStatusAction.java [hunk 3]
```diff
@@ -251,12 +274,17 @@ static ShutdownShardMigrationStatus shardMigrationStatus(
...
             })
-            .map(Tuple::v1)
             .findFirst();
 
-        if (unmovableShard.isPresent()) {
+        if (totalRemainingShards == shardsToIgnoreForFinalStatus.get() && unmovableShard.isEmpty()) {
+            return new ShutdownShardMigrationStatus(
+                SingleNodeShutdownMetadata.Status.COMPLETE,
+                0,
+                "[" + shardsToIgnoreForFinalStatus.get() + "] primary shards cannot be moved away from this node"
```
Comment (dakrone) [inline x-pack/plugin/shutdown/src/main/java/org/elasticsearch/xpack/shutdown/TransportGetShutdownStatusAction.java:283]: Might want a slightly more reassuring message here :)  (also, these aren't all just primary shards) perhaps something like:

`[3] shards cannot be moved away from this node but are replicated to another node in the cluster`

**Review by dakrone — APPROVED** [review]
LGTM, I left some pretty minor comments but nothing major, thanks for fixing this!

File: x-pack/plugin/shutdown/src/internalClusterTest/java/org/elasticsearch/xpack/shutdown/NodeShutdownShardsIT.java (hunk 2 above)
Comment (gwbrown) [reply x-pack/plugin/shutdown/src/internalClusterTest/java/org/elasticsearch/xpack/shutdown/NodeShutdownShardsIT.java:180]: That situation is tested in `TransportGetShutdownStatusActionTests#testStalled`, I also added a unit-test version of this test to that suite as well.

File: x-pack/plugin/shutdown/src/main/java/org/elasticsearch/xpack/shutdown/TransportGetShutdownStatusAction.java (hunk 3 above)
Comment (gwbrown) [reply x-pack/plugin/shutdown/src/main/java/org/elasticsearch/xpack/shutdown/TransportGetShutdownStatusAction.java:283]: Done, this wording is definitely better.

File: x-pack/plugin/shutdown/src/internalClusterTest/java/org/elasticsearch/xpack/shutdown/NodeShutdownShardsIT.java (hunk 1 above)
Comment (gwbrown) [reply x-pack/plugin/shutdown/src/internalClusterTest/java/org/elasticsearch/xpack/shutdown/NodeShutdownShardsIT.java:148]: Thanks!

**Pull request merged**